SoftLayer/CLI/subnet/detail.py

```diff
@@ -25,7 +25,10 @@ def cli(env, identifier, no_vs, no_hardware):
     mgr = SoftLayer.NetworkManager(env.client)
     subnet_id = helpers.resolve_id(mgr.resolve_subnet_ids, identifier,
                                    name='subnet')
-    subnet = mgr.get_subnet(subnet_id)
+
+    mask = 'mask[ipAddresses[id, ipAddress], datacenter, virtualGuests,hardware]'
```

Comment:
add a space between virtualGuests and hardware, or remove all spaces from the mask.